addons/remix-tree-button/main.js

```diff
@@ -1,35 +1,174 @@
-          remixtree.className = "button action-button remixtree-button";
-          remixtree.id = "scratchAddonsRemixTreeBtn";
-          remixtree.appendChild(remixtreeSpan);
-          remixtree.addEventListener("click", () => {
-            window.location.href = `https://scratch.mit.edu/projects/${
-              window.location.href.split("projects")[1].split("/")[1]
-            }/remixtree`;
+    if (document.querySelector("#scratchAddonsRemixTreeBtn") && addon.settings.get("type") === "new") {
+      document.querySelector("#scratchAddonsRemixTreeBtn").style.display = "block";
+      linkOpensInSameTab();
+    } else {
+      if (addon.tab.editorMode === "projectpage") {
+        addon.tab
+          .waitForElement(".flex-row.subactions", {
+            reduxCondition: (state) => state.scratchGui.mode.isPlayerOnly,
+          })
+          .then(() => {
+            if (!document.querySelector(".copy-link-button")) return;
+            if (addon.self.disabled) return;
+            if (addon.settings.get("type") === "replace") {
+              // Change the link from ".../remixes" to ".../remixtree".
+              addon.tab.waitForElement(".remix-list").then(() => {
+                document.querySelector(".remix-list a").href = window.location.href + "remixtree";
+                if (addon.settings.get("new-tab")) {
+                  linkOpensInNewTab();
+                } else {
+                  linkOpensInSameTab();
+                }
+                document.querySelector(".remix-list a span").textContent = msg("remix-tree-link");
+              });
+            } else {
+              // Add a button next to "Copy Link".
+              linkOpensInSameTab();
+              if (document.querySelector("#scratchAddonsRemixTreeBtn")) return; // Check again because we're inside a promise
+              const remixtree = document.createElement("button");
+              const remixtreeSpan = document.createElement("span");
+              remixtreeSpan.innerText = msg("remix-tree");
+              remixtree.className = "button action-button remixtree-button";
+              remixtree.id = "scratchAddonsRemixTreeBtn";
+              remixtree.appendChild(remixtreeSpan);
+              remixtree.addEventListener("click", (e) => {
+                goToRemixTree(e);
+              });
+              addon.tab.appendToSharedSpace({ space: "afterCopyLinkButton", element: remixtree, order: 0 });
+            }
           });
-          addon.tab.appendToSharedSpace({ space: "afterCopyLinkButton", element: remixtree, order: 0 });
-        });
+      }
+    }
+  }
+
+  function goToRemixTree(e) {
+    // Follow the link to the remix tree after clicking the button.
+    if (addon.settings.get("open-with-icon") && !addon.self.disabled) {
+      if (addon.settings.get("new-tab") || e.ctrlKey || e.metaKey) {
+        window.open(
+          `https://scratch.mit.edu/projects/${window.location.href.split("projects")[1].split("/")[1]}/remixtree`,
```

Comment:
Just adding `remixtree` would be enough assuming there's a trailing slash in the URL, but if you want to be sure it would probably be better to store the project ID in a variable.